notebooks/samples/explanations/tf2/ai-explanations-image.ipynb

```diff
@@ -0,0 +1,1746 @@
+{
```

Comment:
<p>[Copying suggestion made in email thread]</p><p> </p><p>Can we use the code from <a href="https://render.githubusercontent.com/view/ipynb?commit=3c4e5e8bdee3061536b96cdc84b980dcd7f62e89&amp;enc_url=68747470733a2f2f7261772e67697468756275736572636f6e74656e742e636f6d2f476f6f676c65436c6f7564506c6174666f726d2f61692d706c6174666f726d2d73616d706c65732f336334653565386264656533303631353336623936636463383462393830646364376636326538392f6e6f7465626f6f6b732f73616d706c65732f6578706c616e6174696f6e732f61692d6578706c616e6174696f6e732d746162756c61722e6970796e62&amp;nwo=GoogleCloudPlatform%2Fai-platform-samples&amp;path=notebooks%2Fsamples%2Fexplanations%2Fai-explanations-tabular.ipynb&amp;repository_id=182338648&amp;repository_type=Repository#Getting-predictions-and-explanations-on-deployed-model" target="_blank">this section of the TF v1 notebooks</a> here? That would change gcloud to the discovery client library and potentially resolve an issue (on both the tabular and image notebooks).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/195/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='75'/>

Reply:
The "Making the explain request" section, to be more clear.

Reply:
I see it now. Perhaps Torry could try this and see if JSON decode error goes away on local laptop:

# This is adapted from a sample in the docs
# Find it here: https://cloud.google.com/ai-platform/prediction/docs/online-predict#python

def predict_json(project, model, instances, version=None):
    """Send json data to a deployed model for prediction.

    Args:
        project (str): project where the AI Platform Model is deployed.
        model (str): model name.
        instances ([Mapping[str: Any]]): Keys should be the names of Tensors
            your deployed model expects as inputs. Values should be datatypes
            convertible to Tensors, or (potentially nested) lists of datatypes
            convertible to tensors.
        version: str, version of the model to target.
    Returns:
        Mapping[str: any]: dictionary of prediction results defined by the
            model.
    """

    service = googleapiclient.discovery.build('ml', 'v1')
    name = 'projects/{}/models/{}'.format(project, model)

    if version is not None:
        name += '/versions/{}'.format(version)

    response = service.projects().explain(
        name=name,
        body={'instances': instances}
    ).execute()

    if 'error' in response:
        raise RuntimeError(response['error'])

    return response